File: sample/src/main/java/uk/co/barbuzz/beerprogressview/sample/PourBeerTask.java
package uk.co.barbuzz.beerprogressview.sample;

import android.os.AsyncTask;

import uk.co.barbuzz.beerprogressview.BeerProgressView;

/**
 * Old school Async to update progress view gradually
 */
public class PourBeerTask extends AsyncTask<Void, Integer, Void> {

    private static final int SLEEP_TIME = 70;
    private final BeerProgressView mBeerProgressView;

    public PourBeerTask(BeerProgressView beerProgressView) {
        mBeerProgressView = beerProgressView;
    }

    @Override
    protected void onProgressUpdate(Integer... values) {
        super.onProgressUpdate(values);
        mBeerProgressView.setBeerProgress(values[0]);
    }

    @Override
    protected Void doInBackground(Void... params) {
        for (int i = 0; i < 90; i++) {
            publishProgress(i);
            try {
                if (i > 10) Thread.sleep(SLEEP_TIME);

            } catch (InterruptedException e) {

            }
            if (isCancelled()) {
                break;
            }
        }
        return null;
    }
}

File: sample/src/main/java/uk/co/barbuzz/beerprogressview/sample/PourBeerTask.kt
package uk.co.barbuzz.beerprogressview.sample

import android.os.AsyncTask

import uk.co.barbuzz.beerprogressview.BeerProgressView

/**
 * Old school Async to update progress view gradually
 */
class PourBeerTask(private val mBeerProgressView: BeerProgressView) : AsyncTask<Void, Int, Void>() {

    override fun onProgressUpdate(vararg values: Int?) {
        super.onProgressUpdate(*values)
        mBeerProgressView.beerProgress = values[0]!!
    }

    override fun doInBackground(vararg params: Void): Void? {
        for (i in 0..90) {
            publishProgress(i)
            try {
                if (i > 10) Thread.sleep(SLEEP_TIME.toLong())

            } catch (e: InterruptedException) {

            }

            if (isCancelled) {
                break
            }
        }
        return null
    }

    companion object {

        private val SLEEP_TIME = 70
    }
}
